optuna/samplers/_grid.py

```diff
@@ -277,3 +277,9 @@ def _same_search_space(self, search_space: Mapping[str, Sequence[GridValueType]]
                     return False
 
         return True
+
+    def is_exhausted(self, study: Study) -> bool:
+        """
+        Returns True if the grid of GridSampler is exhausted, else returns False
```

Comment:
[nit] It is better to describe the function more clearly.
```suggestion
        Return True if all the possible params are evaluated, otherwise return False.
```